### ISSUE TITLE
Could not find a declaration file for module 'kareem'.

### ISSUE THREAD

**Issue by Capta1nRaj** [Automattic/mongoose#14497]
### Prerequisites

- [X] I have written a descriptive issue title

### Mongoose version

8.3.0

### Node.js version

20.x

### MongoDB version

6.0.4

### Operating system

Windows

### Operating system version (i.e. 20.04, 11.3, 10)

11 version 23H2

### Issue

This was my learning project [connect2mongodb](https://www.npmjs.com/package/connect2mongodb). I don't need to create files to connect and disconnect to MongoDB, I just use it.

So, with each update on Mongoose, I update it. However, upon the release of version 8.3.0, whenever I try to npm run build this project, it shows me this error. I am using TypeScript as it was my learning project.

node_modules/mongoose/types/index.d.ts:30:27 - error TS7016: Could not find a declaration file for module 'kareem'. 'E:/Study/Projects/Personal Projects/My Modules/connect2mongodb/node_modules/kareem/index.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/kareem` if it exists or add a new declaration (.d.ts) file containing `declare module 'kareem';`

30   import Kareem = require('kareem');
                             ~~~~~~~~

node_modules/mongoose/types/middlewares.d.ts:2:27 - error TS7016: Could not find a declaration file for module 'kareem'. 'E:/Study/Projects/Personal Projects/My Modules/connect2mongodb/node_modules/kareem/index.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/kareem` if it exists or add a new declaration (.d.ts) file containing `declare module 'kareem';`

2   import Kareem = require('kareem');

It was working fine till 8.2.4.

**Comment by SergeyYahdzhyiev** [comment]
> Same here.
> 
> Reverting back to 8.2.4 fixes it.

Thanks for the tip.

Ran into the same issue.

**Comment by Xirynx** [comment]
Same issue. Also had to revert back to 8.2.4

**Comment by vkarpov15** [comment]
Fixed by #14508

**Comment by Capta1nRaj** [comment]
> Fixed by #14508

Thanks @vkarpov15 for the immediate fix.
